### PR TITLE
Cluster data encryption at rest

### PR DESCRIPTION
**Pull request by mesellings**
## Description

Relates to: https://github.com/camunda/product-hub/issues/1218

Cluster data encryption at rest added, allowing enterprise customers to select a camunda-managed encryption key.

- A new page is added to Components > Concepts > Encryption at rest.
- Link added to this page on the Create a cluster page.

## When should this change go live?

- [ ] This is a bug fix, security concern, or something that needs **urgent release support**.
- [ ] This is already available but undocumented and should be released within a week.
- [ ] This on a **specific schedule** and the assignee will coordinate a release with the DevEx team. (apply `hold` label or convert to draft PR)
- [X] This is part of a scheduled alpha or minor. (apply alpha or minor label)
- [ ] There is **no urgency** with this change and can be released at any time.

## PR Checklist

- [X] My changes are for an **already released minor** and are in `/versioned_docs` directory.
- [X] My changes are for the **next minor** and are in `/docs` directory (aka `/next/`).

- [X] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned the Engineering DRI or delegate.
- [X] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @camunda/tech-writers as a reviewer.

### CODE REVIEW

File: docs/components/concepts/encryption-at-rest.md
Comment (felix-mueller) [inline docs/components/concepts/encryption-at-rest.md:31]: Should we ask customers to reach out if this is not acceptable for them? Perhaps they will anyways so maybe not needed

File: docs/components/concepts/encryption-at-rest.md
Comment (felix-mueller) [inline docs/components/concepts/encryption-at-rest.md:74]: I guess at least anually since it could happen that we need to rotate due to a security incident or similar?

**Review by felix-mueller — APPROVED** [review]
I wonder if there is anything we should mention for Self-Managed encryption. Customers can do this on their own, so wonder if this needs to be mentioned?

@theburi do you have thoughts?

File: docs/components/concepts/encryption-at-rest.md
Comment (Sijoma) [inline docs/components/concepts/encryption-at-rest.md:31]:
```suggestion
Backups use the default provider GCP encryption.
```

This is not something the customer can influence. So we can keep the note concise that the Backups use the other encryption mechanisms.

@felix-mueller would be probably good to at this on our roadmap to encrypt backups too. This will certainly be needed by customer requirements in the end. We already have document handling topic coming up where we could tackle this potentially.

File: docs/components/concepts/encryption-at-rest.md
Comment (Sijoma) [inline docs/components/concepts/encryption-at-rest.md:74]: I would remove the rotation period. We will most likely reduce the time going forward once we learn that we can do this efficiently.

**Review by Sijoma — COMMENTED** [review]
Looks already very well. thank you @mesellings

**Comment by theburi** [comment]
> I wonder if there is anything we should mention for Self-Managed encryption. Customers can do this on their own, so wonder if this needs to be mentioned?
> 
> @theburi do you have thoughts?

I think it would be implied.

File: docs/components/concepts/encryption-at-rest.md
Comment (mesellings) [reply docs/components/concepts/encryption-at-rest.md:74]: Have done, will remove the term "annually" 👍

**Comment by mesellings** [comment]
@Sijoma Do you want to hold this from merging whilst the feature is still being developed, or should we merge it and push the docs live once I've had a TW review? If we do hold for now, we could also add a cropped screenshot to the page to show the three options on the cluster screen once that is complete?

**Comment by Sijoma** [comment]
> @Sijoma Do you want to hold this from merging whilst the feature is still being developed, or should we merge it and push the docs live once I've had a TW review? If we do hold for now, we could also add a cropped screenshot to the page to show the three options on the cluster screen once that is complete?

Lets hold until beginning of next week & when Console was updated with the latest changes. Will ping then here once ready.

**Comment by Sijoma** [comment]
@mesellings when is the next docs release? Can we get this out ~wednesday (18th September)?

**Comment by mesellings** [comment]
> @mesellings when is the next docs release? Can we get this out ~wednesday (18th September)?

Yes of course - I'll see if I can get a TW review before then 👍

**Comment by mesellings** [comment]
@Sijoma Can I just check this is only relevant for 8.6, and doesn't need to be added to the previous doc versions, e.g. 8.5 etc?

**Comment by mesellings** [comment]
@camunda/tech-writers If possible, can we get a review before Wednesday this week, otherwise I will publish and we can do a retro review instead?

File: docs/components/concepts/encryption-at-rest.md
Comment (christinaausley) [inline docs/components/concepts/encryption-at-rest.md:28]: While the UI doesn't read sentence case, we'll have to keep it accordingly -- so, do you mean under **Cluster Details** in **Console**?

File: docs/components/concepts/encryption-at-rest.md
Comment (christinaausley) [inline docs/components/concepts/encryption-at-rest.md:46]: I think since you fully spelled out GCP above, we can keep this as just "GCP encryption".

**Comment by Sijoma** [comment]
> @Sijoma Can I just check this is only relevant for 8.6, and doesn't need to be added to the previous doc versions, e.g. 8.5 etc?

Needs to be back ported to all versions. The feature is version independent. I wonder if we can have this easier on the docs at some point - most of our features don't depend on the version of Camunda. But I guess it's all good since we don't add too many customer-facing features.

**Review by Sijoma — COMMENTED** [review]
nice write up. Thanks @mesellings for making it so easy. 🚀

File: docs/components/concepts/encryption-at-rest.md
Comment (mesellings) [reply docs/components/concepts/encryption-at-rest.md:28]: Thanks - great spot, edited 👍

File: docs/components/concepts/encryption-at-rest.md
Comment (mesellings) [reply docs/components/concepts/encryption-at-rest.md:46]: Thanks - great spot, edited 👍

**Comment by mesellings** [comment]
@Sijoma TW review is complete and I've backported for versions 8.2, 8.3, 8.4, 8.5.

Can I get a re-approval please? I can then merge the PR and publish the docs build today?

**Review by Sijoma — APPROVED** [review]
lgtm

**Pull request merged**
